### PR TITLE
Initial version of the WorldID DRP

### LINKED ISSUE #31: Write a migration plan for the WorldID Identity Manager
No matter how comprehensive the upgradability mechanism is, we need some means to migrate after irrecoverable circumstances. We need to write a migration plan, accounting for at least the following:

- [ ] Restoring the current root.
- [ ] Restoring the root history.
- [ ] Ensuring minimal downtime during a migration.

### PR DESCRIPTION
**Pull request by iamrecursion**
Pretty much what it says on the tin. The DRP is the only non-formatting change.

Closes #31

### CODE REVIEW

File: docs/Disaster Recovery Plan.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+> 1. Obtain write access to the production database from the infrastructure team.
+> 2. [Back up](#roll-back-task-2) the identities and root history tables.
+> 3. [Roll back](#roll-back-task-3) the signup sequencer database such that any identity or root
+>    with `status == "mined"` that is not properly represented on the chain is back in
+>    `status == "pending"`.
+
+Unfortunately it is _possible_ that the signup sequencer record the status of identities and roots
+as `"mined"` when this is spurious (due to on-chain bugs or corruption). To that end we have to be
+able to roll back the database to a state that correctly matches the on-chain state to be restored.
```
Comment (gswirski) [inline docs/Disaster Recovery Plan.md:145]: Is there any way to find the latest root that's common to both the signup-sequencer and the contract?

File: docs/Disaster Recovery Plan.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+> 5. Answer `n` to the question about reusing configuration.
+> 6. Enter the private key obtained from the infrastructure team.
+> 7. Provide `https://polygon-mainnet.g.alchemy.com/v2/ZBjU5pzlBWQQPftvl26hJ4cP1VaBrOGq` as the RPC
+>    URL.
+> 8. Enable the state bridge.
+> 9. Obtain the existing state bridge address.
+> 10. Provide the obtained state bridge address when asked.
+> 11. Leave the "batch insert verifier" address as blank.
+> 12. Set the initial root to the `rootValue` obtained above.
+> 13. Enter the tree depth to match that used by the signup sequencer.
```
Comment (gswirski) [inline docs/Disaster Recovery Plan.md:255]: 14. Store contract addresses in a designated document.

File: docs/Disaster Recovery Plan.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+operations. This may include making calls to functions _other than_ `initialize(...)`. As it
+currently stands, no additional calls need to be made, but this may change due to the updates made
+while performing the [above](#fix-the-problem) tasks.
+
+> #### Deploy: Task 2
+>
+> 1. Obtain the OpenZeppelin relay address.
+> 2. `make transfer-ownership`
+> 3. Answer `n` to the question about reusing configuration.
+> 4. Provide the relay address when asked.
```
Comment (gswirski) [inline docs/Disaster Recovery Plan.md:268]: 4b. Provide new identity manager address.

**Review by gswirski — APPROVED** [review]
Looks great. The biggest question is how to establish the latest valid root. Otherwise I left very minor comments.

File: docs/Disaster Recovery Plan.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+The kinds of issues that can trigger use of this procedure may involve data corruption on chain.
+This means that in order to restore service properly it needs to be very clear what the last known
+good state is.
+
+> #### Record State: Task 1
+>
+> 1. Obtain read access to the production database via the VPN already obtained.
+> 2. Find the row in the signup sequencer database that corresponds to the last correct root on
+>    chain (hereafter `rootValue`). Note that this may not be the same as `latestRoot` as data may
```
[the remark this answers is not in the document]
Comment (iamrecursion) [reply docs/Disaster Recovery Plan.md:100]: It is readable.

File: docs/Disaster Recovery Plan.md (hunk 1 above)
Comment (iamrecursion) [reply docs/Disaster Recovery Plan.md:145]: Unfortunately not. In the presence of data corruption we can basically only rely on the signup sequencer.

**Pull request merged**